contrib/go/src/python/pants/contrib/go/tasks/go_compile.py

```diff
@@ -60,9 +76,28 @@ def execute(self):
           lib_binary_map[vt.target] = os.path.join(gopath, 'pkg', self.goos_goarch,
                                                    vt.target.import_path + '.a')
 
-  def _go_install(self, target, gopath):
-    build_flags = re.sub(r'^"|"$', '', self.get_options().build_flags)
-    args = safe_shlex_split(build_flags) + [target.import_path]
+  @classmethod
+  def _get_build_flags(cls, target, build_flags_from_option, is_flagged):
```

Comment:
You might:
```python
@classmethod
@memoized_method
def _get_build_flags(cls, target, build_flags_from_option, is_flagged):
```

This will get called twice for each invalid target, once by the fp strategy, once by the work loop in execute.